### PR TITLE
Rename AnsibleEE module to point to openstack-k8s-operators

### PR DESCRIPTION
**Pull request by fmount**
This patch introduces a few useful changes to the ansible operator:

1. Bump to `go 1.18` and update the go.{mod, sum} accordingly
2. Rename the main golang module to point to the `openstack-k8s-operators` namespace
3. Add the API module to align the operator to the rest of the openstack operators: this change allows to make the `AnsibleEE` CRD consumable by other projects (e.g., the meta-operator) and gives the ability to add basic `ext-data-plane` triggering support to the top level CRD

Signed-off-by: Francesco Pantano <fpantano@redhat.com>

### CODE REVIEW

**Comment by abays** [comment]
I think this also requires a refactoring to the `Dockerfile`.  When I try to build this, the `Dockerfile` hits a snag:

```
[1/2] STEP 5/9: RUN go mod download
go mod download: github.com/openstack-k8s-operators/ansibleee-operator/api@v0.0.0-00010101000000-000000000000 (replaced by ./api): reading api/go.mod: open /workspace/api/go.mod: no such file or directory
```

It might also require a `templates` dir (even if empty for now) to work with the refactored `Dockerfile`.  We use this new format for the `Dockerfile` in other operators, such as https://github.com/openstack-k8s-operators/glance-operator/blob/master/Dockerfile.

**Comment by fmount** [comment]
> I think this also requires a refactoring to the `Dockerfile`. When I try to build this, the `Dockerfile` hits a snag:
> 
> ```
> [1/2] STEP 5/9: RUN go mod download
> go mod download: github.com/openstack-k8s-operators/ansibleee-operator/api@v0.0.0-00010101000000-000000000000 (replaced by ./api): reading api/go.mod: open /workspace/api/go.mod: no such file or directory
> ```
> 
> It might also require a `templates` dir (even if empty for now) to work with the refactored `Dockerfile`. We use this new format for the `Dockerfile` in other operators, such as https://github.com/openstack-k8s-operators/glance-operator/blob/master/Dockerfile.

Right, thanks for pointing that out. I'll make those changes according to [1]

[1]  https://github.com/openstack-k8s-operators/docs/blob/main/new_operator.md#update-dockerfile

**Comment by fmount** [comment]
> > I think this also requires a refactoring to the `Dockerfile`. When I try to build this, the `Dockerfile` hits a snag:
> > ```
> > [1/2] STEP 5/9: RUN go mod download
> > go mod download: github.com/openstack-k8s-operators/ansibleee-operator/api@v0.0.0-00010101000000-000000000000 (replaced by ./api): reading api/go.mod: open /workspace/api/go.mod: no such file or directory
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It might also require a `templates` dir (even if empty for now) to work with the refactored `Dockerfile`. We use this new format for the `Dockerfile` in other operators, such as https://github.com/openstack-k8s-operators/glance-operator/blob/master/Dockerfile.
> 
> Right, thanks for pointing that out. I'll make those changes according to [1]
> 
> [1] https://github.com/openstack-k8s-operators/docs/blob/main/new_operator.md#update-dockerfile

@abays does this looks better now?

File: bundle.Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=ansibleee-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.22.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-unknown
```
Comment (abays) [inline bundle.Dockerfile:9]: This seems odd.  What version of operator SDK were you using?

Comment (fmount) [reply bundle.Dockerfile:9]: mmm ok I suspect the problem there is the fact I updated my system and it got the new sdk version:

```
> s pacman -Qi operator-sdk
Name            : operator-sdk
Version         : 1.25.0-1
Description     : SDK for building Kubernetes applications
Architecture    : x86_64
URL             : https://github.com/operator-framework/operator-sdk
```

I'll try to `make bundle` using the tool provided by [1].
Quick question: should this file (bundle.Dockerfile) be omitted here? It's generated by that command but I suspect it shouldn't be part of the PR (and generated in a different context).

[1] https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/roles/download_tools/tasks/main.yaml#L31-L48

Comment (abays) [reply bundle.Dockerfile:9]: Yeah, we don't need to store that file in the repo.  It looks like maybe `.gitignore` is missing some content that we set in other operators, for instance: https://github.com/openstack-k8s-operators/cinder-operator/blob/60bdac1a3f2db7cd1ea1e8a43a3309bc3b4209e7/.gitignore#L20-L23

Comment (fmount) [reply bundle.Dockerfile:9]: ack thanks, I'm going to add that section in `.gitignore`

**Review by abays — APPROVED** [review]
/lgtm

**Review by fultonj — APPROVED** [review]
I don't see any issues here, it /lgtm but I haven't yet tested it (just read). I accidentally said I tested did but I was referring to lib-common PR88. We should get feedback from @jlarriba too.

**Comment by jlarriba** [comment]
lgtm, thank you very much @abays and @fmount! This is awesome.

**Pull request merged**